x-pack/plugins/infra/server/lib/alerting/metric_threshold/lib/create_bucket_selector.ts

```diff
@@ -32,12 +32,14 @@ export const createBucketSelector = (
   const isCount = condition.aggType === Aggregators.COUNT;
   const isRate = condition.aggType === Aggregators.RATE;
   const bucketPath = isCount
-    ? 'currentPeriod>_count'
+    ? "currentPeriod['all']>_count"
```

Comment:
Why do we need to add `['all']` here? I didn't understand how this part works and why previously we didn't need `['all']`, and now we do.

Reply:
For the `count` with KQL filter, we need to use a `filter` aggregation. One of the requirements for the `filter` aggregation is that it needs to be a child of a multi-bucket aggregation. `currentPeriod` was originally implemented as a `filter` aggregation (single bucket), with this change I converted it to a `filters` aggregation  with one bucket defined as `all`. I used `all` because it represents everything (all hits) and I also used it for the non-group by top level multi-bucket (to support `currentPeriod`) so it's consistent throughout.